src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs

```diff
@@ -48,17 +51,66 @@ public async Task Invoke(MessageContext messageContext, CancellationToken cancel
 
                     ex.Data["Pipeline canceled"] = transportReceiveContext.CancellationToken.IsCancellationRequested;
 
+                    // TODO: Add an explicit tag for operation canceled
```

Comment:
It would clarify the state of the trace without looking at the exception details i guess. But the tag could also allow for easy querying to exclude these types of exceptions, or look exactly for these types of exceptions which might be interesting in certain sceanrios.